tmt/utils/filesystem.py

```diff
@@ -0,0 +1,113 @@
+"""
+Utility functions for filesystem operations.
+"""
+
+import os
+import shutil
+
+import tmt.log
+from tmt._compat.pathlib import Path
+from tmt.utils import Command, GeneralError, RunError
+
+
+def _copy_tree_reflinks(
+    src: Path,
+    dst: Path,
+    logger: 'tmt.log.Logger',
+    workdir_root: Path,  # Keep signature consistent, even if not used here
+) -> bool:
+    """
+    Attempt to copy directory using reflinks.
+    """
+    logger.debug(f"Attempting reflink copy from '{src}' to '{dst}'")
+
+    try:
+        # Use cp with --reflink=auto which falls back to regular copy if reflinks not supported
+        # The '/./' at the end of the source path tells cp to copy the *contents* of the directory
+        # rather than creating a new subdirectory in the destination
+        Command('cp', '-a', '--reflink=auto', f"{src}/./", str(dst)).run(
+            cwd=None, logger=logger, join=True, silent=True
+        )
+        return True
+    except RunError as error:
+        # Specific error for command failure (e.g. reflink not supported)
+        logger.debug(f"Reflink copy command failed: {error}, falling back")
+        return False
+    # Let other exceptions (e.g. permissions, disk full) propagate
+
+
+def _copy_tree_basic(
+    src: Path,
+    dst: Path,
+    logger: 'tmt.log.Logger',
+    workdir_root: Path,
+) -> None:
+    """
+    Perform a basic recursive copy of a directory tree.
+
+    Handles files, directories, and symlinks using standard shutil operations.
+    Raises exceptions on failure.
+    """
+    logger.debug(f"Performing basic copy from '{src}' to '{dst}'")
+
+    copied_count = 0
+    for item in src.rglob('*'):
+        relative_path = item.relative_to(src)
+        dst_item = dst / relative_path
```

Comment:
Having `item` and `dst_item` - would you mind renaming `item` to `src_item`? We have a special name for *dst* item, but not for *src* item, that really does not feel right.